reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/completable/TestAwait.kt

```diff
@@ -0,0 +1,32 @@
+package com.badoo.reaktive.test.completable
+
+import com.badoo.reaktive.completable.Completable
+import com.badoo.reaktive.completable.blockingAwait
+
+/**
+ * This method is used when you need to wait for an asynchronous operation to finish
+ * and you can't use or don't want to use the [TestScheduler][com.badoo.reaktive.test.scheduler.TestScheduler].
+ *
+ * It is JavaScript friendly, it returns `Promise` which is when returned from a test method
+ * causes the test to wait for completion. In all other targets [Completable.blockingAwait()][blockingAwait] is used.
+ *
+ * Example:
+ * ```
+ * class Calculator {
+ *     fun sum(a: Int, b: Int): Single<Int> =
+ *         singleFromFunction { a + b }
+ *             .subscribeOn(computationScheduler)
+ * }
+ *
+ * class CalculatorTest {
+ *     @Test
+ *     fun sum_2_3_returns_5() =
+ *         Calculator()
+ *             .sum(2, 3)
+ *             .doOnBeforeSuccess { assertEquals(6, it) }
```

Comment:
This part looks bad. Can't we do something like in code below?

```kotlin
Calculator()
    .sum(2, 3)
    .test()
    .await() // extension on test subscriber, not on Completable or whatever else
    .assertValue(6)
```

Or I am missing the point?

Reply:
Good point. How about the following?

`expect fun <T> Single<T>.testAwait(onSuccess: (T) -> Unit = {})`

```kotlin
class MyLogicTest {

    @Test
    fun foo() =
        MyLogic()
            .sum(1, 2)
            .testAwait { assertEquals(3, it) }
}
```

Reply:
But is not it impossible to check errors in your case? Let's return from `testAwait()` test subscriber?

Reply:
We must return `Promise` for JS. This special case is recognized by the framework and it will wait untill the `Promise` resolves or fails.

Reply:
If you need to verify error. Then the error should be a positive case. Like:

```kotlin
   @Test
    fun foo() =
        MyLogic()
            .div(5, 0)
            .onErrorReturn { it }
            .testAwait { assertTrue(it is ArithmeticException) }
```

Reply:
Or we could add additional argument:
`expect fun <T> Single<T>.testAwait(assertSuccess: (T) -> Unit = {}, assertError: ((Throwable) -> Unit)? = null)`

Reply:
```kotlin
    @Test
    fun foo() =
        MyLogic()
            .div(5, 0)
            .testAwait(assertError = { assertTrue(it is ArithmeticException) }, assertSuccess = { fail() })
```

Reply:
Looks sweet, I like the `assertError` is optional since we never use `onError`, early on the pipeline we convert it to an ADT, like in `Single<Either<ErrorADT, T>`

Reply:
Yeah. The idea of the optional `assertError` callback is: if it is provided then it must fail to make the test failed. If `assertError` succeeds then the error is considered as expected.